pkg/operator/ceph/cluster/osd/health.go

```diff
@@ -184,3 +197,28 @@ func (m *OSDHealthMonitor) restartOSDIfStuck(osdID int) error {
 	}
 	return nil
 }
+
+// updateCephStorage udates the CR with deviceclass details
+func (m *OSDHealthMonitor) updateCephStorage(devices []string) {
+	var err error
+	cephCluster := &cephv1.CephCluster{}
+	cephClusterStorage := cephv1.CephStorage{}
+
+	for _, device := range devices {
+		cephClusterStorage.DeviceClasses = append(cephClusterStorage.DeviceClasses, cephv1.DeviceClasses{Name: device})
+	}
+	err = m.context.Client.Get(context.TODO(), m.clusterInfo.NamespacedName(), cephCluster)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debug("CephCluster resource not found. Ignoring since object must be deleted.")
+			return
+		}
+		logger.Errorf("failed to retrieve ceph cluster %q to update ceph Storage. %v", m.clusterInfo.NamespacedName().Name, err)
+		return
+	}
+	cephCluster.Status.CephStorage = &cephClusterStorage
+	if err := opcontroller.UpdateStatus(m.context.Client, cephCluster); err != nil {
```

Comment:
Can we skip calling `UpdateStatus()` if the list hasn't changed?

Reply:
Implemented check for change in list before calling `UpdateStatus()`